From c8ab551c0c2a84b4a0d758d13a1b00d8f9f1864d Mon Sep 17 00:00:00 2001
From: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
Date: Sat, 1 Jul 2023 10:05:50 +0530
Subject: [PATCH 3/9] wifi: ath12k: war: Fix the crash during associate 6G slo

Due to the station's 6GHz power mode being unset(0), the station crashes
while trying to acquire a 6GHz psd channel. since the get_psd_channel
contains code to transform reg_6g_power_mode 0 based indexing for AP.

The workaround added to aviod the crash, The patch should be remove with
potential fix.

Signed-off-by: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -12553,9 +12553,11 @@ void ath12k_mac_fill_reg_tpc_info(struct
         * AP cant parse its own beacon. Hence, we get the 6g power mode
         * from the wdev corresponding to the struct ieee80211_vif
         */
-       if (ahvif->vdev_type == WMI_VDEV_TYPE_STA)
+       if (ahvif->vdev_type == WMI_VDEV_TYPE_STA) {
                reg_6g_power_mode = bss_conf->power_type;
-       else if (ahvif->vdev_type == WMI_VDEV_TYPE_AP) {
+		if (reg_6g_power_mode == IEEE80211_REG_UNSET_AP)
+			reg_6g_power_mode = IEEE80211_REG_LPI_AP;
+	} else if (ahvif->vdev_type == WMI_VDEV_TYPE_AP) {
                struct wireless_dev *wdev = ieee80211_vif_to_wdev(vif);
                /* With respect to ieee80211, the 6G AP power mode starts from index
                 * 1 while the power type stored in struct wireless_dev is based on
